compiler/rustc_interface/src/passes.rs

```diff
@@ -273,7 +273,7 @@ fn configure_and_expand(
 
     // Done with macro expansion!
 
-    resolver.resolve_crate(&krate);
+    resolver.resolve_crate(&mut krate);
```

Comment:
I think the new pass is supposed to run right after this function *finishes*, at least for the step 1 in https://github.com/rust-lang/rust/issues/99669#issuecomment-1193308590.
Also makes sense to put this pass into a separate file.
(I'm still not sure it's a good idea, all in all, maybe it will become more visible once the step 1 is finished.)